doc/fgpu.design.rst

```diff
@@ -177,51 +177,132 @@ slow and require multiple passes over the memory, because
+be the number of channels into which we wish to decompose the input sequence into, and let
+:math:`x_i` be the (real) time-domain samples (:math:`0 \le i < 2N`) and :math:`X_k` be its discrete
+Fourier transform (DFT). Because the time domain is real, the frequency domain is Hermitian symmetric,
+and we only need to compute half of it to recover all the information. We thus only need to consider
+:math:`k` from :math:`0` to :math:`N-1` (this loses information about :math:`X_N`, but it is convenient
+to discard it and thus have a power-of-two number of outputs).
+
+.. math:: X_k = \sum_{i=0}^{2N-1} e^{\frac{-2\pi j}{2N}\cdot ik} x_i.
+
+We know that a direct implementation of the DFT is inefficient and alternative, more efficient
+means exist to perform this computation. One such method is the FFT introduce by Cooley-Tukey
+and in the GPU space cuFFT is one such implementation. As highlighted earlier, transform sizes
+of greater than 16384 (for a GeForce RTX 3080 Ti at least) require more than one memory pass making it
+less efficient than it needs to be. The technique detailed below uses the decomposition as provided by
+Cooley-Tukey to break down a larger transform into smaller 'sub-transforms' where the number of 'sub-transforms'
+is intentionally kept small for efficiency reasons and later combined (same process as the FFT) to form
+the larger transform size. This is a multi-step process and requires some extra notation and math tricks.
 
 Real-to-complex transform
 ~~~~~~~~~~~~~~~~~~~~~~~~~
-Let's introduce some notation to see how this works. Let :math:`n` be the
-number of channels, :math:`x_i` be the (real) time-domain samples and
-:math:`X_k` be the Fourier transform (we divide by :math:`2n` rather than
-:math:`n` because that's the number of input samples used).
+Now for some notation to see how this works. We start by treating :math:`x` (a real array of length N) as if it is
+a complex array :math:`z` of length N, with each adjacent pair of real values in :math:`x` interpreted as the
+real and imaginary components of a complex value, and computing the Fourier transform of :math:`z`. Formally,
+let :math:`u_i = x_{2i}` and :math:`v_i = x_{2i+1}`. Then :math:`z_i = u_i + jv_i = x_{2i} + j x_{2i+1}`.
+
+For clarity, array :math:`x` has length :math:`2N` but is divided into even- and odd- samples
+each of length :math:`N`. In other words, each adjacent pair of real values in :math:`x` is interpreted as the real
+and imaginary components of a complex value.
+
+We will start by computing the Fourier transform of :math:`z`. This is convenient to do because :math:`z` has the
+identical memory layout to :math:`x`, so we can obtain it just by viewing the same memory as containing N complex
+numbers instead of 2N real numbers. In other words, each adjacent pair of real values in :math:`x` is interpreted
+as the real and imaginary components of a complex value.
```

Comment:
And you can delete everything after the first sentence in this paragraph too (as well as the paragraph break at the end).